component/githosts.jsonnet

```diff
@@ -0,0 +1,21 @@
+// main template for lieutenant
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+local inv = kap.inventory();
+// The hiera parameters for the component
+local params = inv.parameters.lieutenant;
+
+{
+  ['%s' % [ std.asciiLower(git.name) ]]: kube.Secret(git.name) {
+    metadata: {
+      name: git.name,
```

Comment:
you never need to set `metadata.name` when using kube-libsonnet as that field is set to the value provided as parameter to `kube.Secret()` (in this instance). Instead just make sure you use 

```
metadata+: {
  namespace: params.namespace,
},
```